test/PowerPod.js

```diff
@@ -1,60 +1,154 @@
+
+        const stakeAndRegisterInDelegation = async (st1inch, delegation, user, amount, userIndex) => {
+            await st1inch.connect(user).deposit(0, commonLockDuration);
+            await st1inch.depositFor(user.address, amount);
+            await st1inch.connect(user).addPod(delegation.address);
+            await delegation
+                .connect(user)
+                .functions['register(string,string)'](
+                    `${userIndex}DelegatingToken`,
+                    `A${userIndex}DT`,
+                );
+            await delegation.connect(user).delegate(user.address);
+        };
+
+        const depositAndDelegateTo = async (st1inch, delegation, from, to, amount, duration = commonLockDuration) => {
+            await st1inch.connect(from).deposit(amount, duration);
+            await st1inch.connect(from).addPod(delegation.address);
+            await delegation.connect(from).delegate(to);
+        };
 
-        const St1inch = await ethers.getContractFactory('St1inch');
-        const st1inch = await St1inch.deploy(oneInch.address, expBase, addr.address);
-        await st1inch.deployed();
+        const accounts = await ethers.getSigners();
+        const [owner, alice] = accounts;
+
+        const oneInch = await deployContract('ERC20PermitMock', ['1inch', '1inch', owner.address, ether('200')]);
+        await oneInch.transfer(alice.address, ether('100'));
+
+        const st1inch = await deployContract('St1inch', [oneInch.address, expBase, owner.address]);
         await oneInch.approve(st1inch.address, ether('100'));
+        await oneInch.connect(alice).approve(st1inch.address, ether('100'));
 
-        const PowerPod = await ethers.getContractFactory('PowerPod');
-        const delegation = await PowerPod.deploy('PowerPod', 'PP', st1inch.address);
-        await delegation.deployed();
-        await stakeAndRegisterInDelegation(st1inch, delegation, addr1, amountToStake, 0);
+        const delegation = await deployContract('PowerPod', ['PowerPod', 'PP', st1inch.address]);
 
-        return { st1inch, delegation };
-    }
+        const whitelistRegistry = await deployContract('WhitelistRegistry', [delegation.address, BALANCE_THRESHOLD]);
+        // fill all whitelist into WhitelistRegistry
+        for (let i = 0; i < MAX_WHITELISTED; ++i) {
+            const userIndex = i + 2;
+            const user = accounts[userIndex];
+            await stakeAndRegisterInDelegation(st1inch, delegation, user, ether('2') * BigInt(i + 1), userIndex);
+            await whitelistRegistry.connect(user).register();
+        }
+        await stakeAndRegisterInDelegation(st1inch, delegation, owner, ether('1'), 0);
 
-    before(async function () {
-        accounts = await ethers.getSigners();
-        addr = accounts[0];
-        addr1 = accounts[1];
-    });
+        return {
+            contracts: { st1inch, delegation, whitelistRegistry },
+            accounts: { owner, alice, accounts },
```

Comment:
That's too much, if we unwrap accounts, we can return only the unwrapped ones like `accounts: { owner, alice, bob, ... }`